src/Tickets/Seating/Service/OAuth_Token.php

```diff
@@ -24,19 +29,21 @@ trait OAuth_Token {
 	 *
 	 * @return string|null The OAuth token, or `null` if there is no OAuth token.
 	 */
-	private function get_oauth_token(): ?string {
-		return tribe_get_option( self::get_oauth_token_option_name(), null );
-	}
+	protected function get_oauth_token(): ?string {
+		$memoize      = tribe_cache();
+		$cache_key    = 'tec_seating_access_token';
+		$access_token = $memoize[ $cache_key ];
 
-	/**
-	 * Returns the option name used to store the Service oAuth access token.
-	 *
-	 * @since TBD
-	 *
-	 * @return string
-	 */
-	public static function get_oauth_token_option_name(): string {
-		return 'events_tickets_seating_access_token';
+		if ( ! ( $access_token && is_string( $access_token ) ) ) {
+			try {
+				$access_token = get_authorization_token( 'tec-seating' );
+			} catch ( RuntimeException $e ) {
+				$access_token = null;
+			}
+			$memoize[ $cache_key ] = $access_token;
+		}
+
+		return $access_token;
```

Comment:
If the access token is not set, i.e. the user did not connect, then the value returned at line `39` will be `null`.
The `catch` block is replicating that to handle errors.

If the access token is `null`, then it should not be considered  a valid value, hence the `is_string` check, and it should be refetched from storage.
This will allow read-write-read requests to work correctly, without requiring the code updating the token to directly or indirectly invalidate some caches.

The storage of the token is in-memory and either an option (current implementation) or a transient (alt. implementation); those will be loaded in memory and read from there after a first access making them fast refetches that will not map to db queries on each in-request access.

For the above reasons, I would leave the code as is.